src/icalendar/cal.py

```diff
@@ -587,6 +588,74 @@ def is_thunderbird(self) -> bool:
         return any(attr.startswith("X-MOZ-") for attr in self.keys())
 
 
+    @classmethod
+    def from_file(cls, file: Union[str, Path], multiple: bool = False):
+        """Create a Component from a file.
+
+        This class method can be used by any Component subclass (Calendar, Event, etc.)
+        to read their data from a file.
+
+        Args:
+            file: The file to read from. Can be:
+                - A string path to a file
+                - A Path object
+            multiple: If True, allows parsing multiple components from the file.
+
+        Returns:
+            If multiple=False (default):
+                A single Component instance of the appropriate type
+            If multiple=True:
+                A list of Component instances
+
+        Raises:
+            FileNotFoundError: If the file path doesn't exist
+            ValueError: If the file contents are not valid iCalendar format
+
+        Example:
+            >>> from icalendar import Calendar
+            >>> # Read a calendar file
+            >>> cal = Calendar.from_file("src/icalendar/tests/calendars/example.ics")
+            >>> # Read multiple calendars
+            >>> cals = Calendar.from_file("src/icalendar/tests/calendars/multiple_calendar_components.ics", multiple=True)
+        """
+        # Handle string path by converting to Path
+        if isinstance(file, str):
+            file = Path(file)
+
+        return cls.from_ical(file.read_bytes(), multiple=multiple)
+
+    def to_file(self, file: Union[str, Path], sorted: bool = True) -> None:
+        """Write the component to a file.
+
+        This method can be used by any Component subclass (Calendar, Event, etc.)
+        to write their data to a file.
+
+        Args:
+            file: Where to write the component. Can be:
+                - A string path to a file
+                - A Path object
+            sorted: Whether parameters and properties should be lexicographically sorted.
+
+        Example:
+            >>> from icalendar import Calendar
+            >>> from pathlib import Path
+            >>> # Read a calendar file
+            >>> cal = Calendar.from_file("src/icalendar/tests/calendars/example.ics")
```

Comment:
The examples here seem to be the same as above. You can use a temporary file for this.

Reply:
I changed the examples to use the `to_file()` method